tests/unit/synapseutils/unit_test_synapseutils_copy.py

```diff
@@ -41,4 +41,56 @@ def test_copyWiki_input_validation():
         mock_getWiki.assert_has_calls(expected_calls)
 
         assert_raises(ValueError, synapseutils.copyWiki, syn, "syn123", "syn456", entitySubPageId="some_string",
-                              updateLinks=False)
\ No newline at end of file
+                              updateLinks=False)
+
+
+def test_copyFileHandles_input_validation():
+    file_handles = ["test"]
+    obj_types = []
+    obj_ids = [None]
+    assert_raises(ValueError, synapseutils.copyFileHandles, syn, file_handles, obj_types, obj_ids)
+
+
+class TestCopyFileHandles:
+
+    def setup(self):
+        self.patch_restPOST = patch.object(syn, 'restPOST')
+        self.mock_restPOST = self.patch_restPOST.start()
+
+    def teardown(self):
+        self.patch_restPOST.stop()
+
+    def test_copy_invalid_input(self):
+        file_handles = ["test"]
+        obj_types = []
+        obj_ids = [None]
+        assert_raises(ValueError, synapseutils.copyFileHandles, syn, file_handles, obj_types, obj_ids)
+
+    def test_copy_two_files(self):
+        file_handles = ["40827299", "41414463"]
+        obj_types = ["FileEntity", "FileEntity"]
+        obj_ids = ["20181837", "20459234"]
+        con_types = [None, "wiki"]
+        file_names = [None, "test"]
+        synapseutils.copyFileHandles(syn, file_handles, obj_types, obj_ids, con_types, file_names)
+        expected = {
+                        "copyRequests": [
+                            {
+                                "originalFile": {
+                                    "fileHandleId": "40827299",
+                                    "associateObjectId": "20181837",
+                                    "associateObjectType": "FileEntity"
+                                }
+                            },
+                            {
+                                "originalFile": {
+                                    "fileHandleId": "41414463",
+                                    "associateObjectId": "20459234",
+                                    "associateObjectType": "FileEntity"
+                                },
+                                "newContentType": "wiki",
+                                "newFileName": "test"
+                            }
+                        ]
+                    }
+        self.mock_restPOST.assert_called_once_with('/filehandles/copy', body=json.dumps(expected), endpoint=syn.fileHandleEndpoint)
```

Comment:
test cases:
1. mock that the POST response contains mix results including success and failure
2. mock case where all items fit in 1 call
3. mock case where all items got split into 2 calls, and the results are combined.